aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -261,62 +262,72 @@ static void cauchy_kernel(TensorIterator& iter, double median, double sigma, c10
+    int64_t seed;
+    {
+      // See Note [Acquire lock when using random generators]
+      std::lock_guard<std::mutex> lock(generator->mutex_);
+      seed = generator->random();
     }
+    int64_t n = self.numel();
+    bool contig = self.is_contiguous();
+
+    AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Bool, self.scalar_type(), "bernoulli_scalar_cpu_", [&] {
+      at::Tensor tmp_int_tensor;
+      if (std::is_same<scalar_t, int>::value && contig) {
+        tmp_int_tensor = self;
+      } else {
+        tmp_int_tensor = at::empty(self.sizes(), self.options().dtype(at::kInt));
+      }
 
-    scalar_t *self_ptr = self.data_ptr<scalar_t>();
-    int *sample_int_ptr = tmp_int_tensor.data_ptr<int>();
-
-    auto sample = [&](int64_t begin, int64_t end) {
-      int64_t len = end - begin;
-      if (len > 0) {
-        VSLStreamStatePtr stream;
-        vslNewStream(&stream, VSL_BRNG_MCG31, seed);
-        vslSkipAheadStream(stream, begin);
-        viRngBernoulli(VSL_RNG_METHOD_BERNOULLI_ICDF, stream, len,
-          sample_int_ptr + begin, p);
-        vslDeleteStream(&stream);
-
-        // vectorized copy if using buffer and contiguous, i.e., being non-int
-        // type and contiguous
-        if (!std::is_same<scalar_t, int>::value && contig) {
-          scalar_t *self_seg = self_ptr + begin;
-          int* tmp_seg = sample_int_ptr + begin;
-          at::vec256::convert<int, scalar_t>(tmp_seg, self_seg, len);
+      scalar_t *self_ptr = self.data_ptr<scalar_t>();
+      int *sample_int_ptr = tmp_int_tensor.data_ptr<int>();
+
+      auto sample = [&](int64_t begin, int64_t end) {
+        int64_t len = end - begin;
+        if (len > 0) {
+          VSLStreamStatePtr stream;
+          vslNewStream(&stream, VSL_BRNG_MCG31, seed);
+          vslSkipAheadStream(stream, begin);
+          viRngBernoulli(VSL_RNG_METHOD_BERNOULLI_ICDF, stream, len,
+            sample_int_ptr + begin, p);
+          vslDeleteStream(&stream);
+
+          // vectorized copy if using buffer and contiguous, i.e., being non-int
+          // type and contiguous
+          if (!std::is_same<scalar_t, int>::value && contig) {
+            scalar_t *self_seg = self_ptr + begin;
+            int* tmp_seg = sample_int_ptr + begin;
+            at::vec256::convert<int, scalar_t>(tmp_seg, self_seg, len);
+          }
         }
-      }
-    };
+      };
 
-    parallel_for(0, n, /* grain_size= */ 800, sample);
+      parallel_for(0, n, /* grain_size= */ 800, sample);
 
-    // copy_ if using buffer and non contiguous
-    if (!contig) {
-      self.copy_(tmp_int_tensor);
-    }
-  });
+      // copy_ if using buffer and non contiguous
+      if (!contig) {
+        self.copy_(tmp_int_tensor);
+      }
+    });
+  } else {
+    // Use AT_ASSERTM because this should never be reached, and AT_ASSERTM tells
+    // users to report this as a bug.
+    AT_ASSERTM(false, "ATen not compiled with MKL");
```

Comment:
Is this intended even though we are now in the preprocessor branch where `AT_MKL_ENABLED()` is `TRUE`?